trillian/ctfe/ct_server/main.go

```diff
@@ -130,17 +131,19 @@ func main() {
 			glog.Exitf("Failed to create etcd metrics manager: %v", err)
 		}
 
-		// Another option would be to have a single service and two endpoints "http" and "metrics".
-		glog.Infof("Announcing our presence in %v with %+v", *etcdHTTPService, *httpEndpoint)
-		httpManager.AddEndpoint(ctx, *etcdHTTPService+"/", endpoints.Endpoint{Addr: *httpEndpoint})
+		etcdHTTPKey := fmt.Sprintf("%s/%s", *etcdHTTPService, *httpEndpoint)
+		glog.Infof("Announcing our presence at %v with %+v", etcdHTTPKey, *httpEndpoint)
+		httpManager.AddEndpoint(ctx, etcdHTTPKey, endpoints.Endpoint{Addr: *httpEndpoint})
```

Comment:
If you still consider squashing things into 1 service/manager, I think this is how it could look like:

```golang
m, err := endpoints.NewManager(client, "ctfe")
m.AddEndpoint(ctx, "ctfe/http/<*httpEndpoint>", endpoints.Endpoint{Addr: *httpEndpoint})
m.AddEndpoint(ctx, "ctfe/metrics/<*metricsEndpoint>", endpoints.Endpoint{Addr: *metircsEndpoint})
```

That wouldn't be backward compatible though.

Reply:
I'm all about that backwards compatibility. I only made a change because I thought I was compelled to. Luckily, I don't. This command/output is the same before/after this change:

```
etcdctl get trillian-ctfe --prefix
trillian-ctfe-http/localhost:6965
{"Op":0,"Addr":"localhost:6965","Metadata":null}
trillian-ctfe-metrics-http/localhost:6965
{"Op":0,"Addr":"localhost:6965","Metadata":null}
```